base/base_tests/non_intersecting_intervals_tests.cpp

```diff
@@ -0,0 +1,58 @@
+#include "testing/testing.hpp"
+
+#include "base/non_intersecting_intervals.hpp"
+
+namespace
+{
+using namespace base;
+
+UNIT_TEST(NonIntersectingIntervals_1)
+{
+  NonIntersectingIntervals<int> intervals;
+
+  TEST(intervals.AddInterval(1, 10), ());
+  TEST(intervals.AddInterval(11, 15), ());
+  // Overlap with [1, 10].
+  TEST(!intervals.AddInterval(1, 20), ());
+
+  // Overlap with [11, 15].
+  TEST(!intervals.AddInterval(13, 20), ());
+
+  // Overlap with [1, 10] and [11, 15].
+  TEST(!intervals.AddInterval(0, 100), ());
+
+  TEST(intervals.AddInterval(100, 150), ());
+
+  // Overlap with [100, 150].
+  TEST(!intervals.AddInterval(90, 200), ());
+}
+
+UNIT_TEST(NonIntersectingIntervals_2)
+{
+  NonIntersectingIntervals<int> intervals;
+
+  TEST(intervals.AddInterval(1, 10), ());
+  // Overlap with [1, 10].
+  TEST(!intervals.AddInterval(2, 9), ());
+}
+
+UNIT_TEST(NonIntersectingIntervals_3)
+{
+  NonIntersectingIntervals<int> intervals;
+
+  TEST(intervals.AddInterval(1, 10), ());
+  // Overlap with [1, 10].
+  TEST(!intervals.AddInterval(0, 20), ());
+}
+
+UNIT_TEST(NonIntersectingIntervals_4)
```

Comment:
Please add a test similar to [0, 1], [1, 2], [2, 3], [4, 5], [1,3], [2, 2], [2, 5].
It is hidden in the current interface but it is a test against a possible implementation where adjacent intervals are coalesced (as we do in https://github.com/mapsme/omim/blob/master/search/interval_set.hpp)

Reply:
add